Add Support for index.juvix.md and README.juvix.md as Homepage

agureev opened the pull request:
This PR closes:
- https://github.com/anoma/juvix-mkdocs/issues/2

It updates the plugin file to include `on_nav` function which sets `index.juvix.md` file as homepage in navigation if detected and chnages `on_page_markdown` to reference any such files without `/index/` reference when building.

In practical terms, sets `index.juvix.md` file as the homepage when detected.

agureev committed:
Add Support for index.juvix.md as Homepage

Updates the plugin file to include `on_nav` function which sets
`index.juvix.md` file as homepage in navigation if detected and
chnages `on_page_markdown` to reference any such files without
`/index/` reference when building.

In practical terms, sets `index.juvix.md` file as the homepage when detected.

jonaprieto commented on mkdocs_juvix/plugin.py:
```diff
@@ -82,10 +82,26 @@ def on_page_read_source(self, page: Page, config):
-                ".juvix", "")
+                text, "")
+
+        if page.file.name.endswith(index):
```
I'd say the condition here is `== index`.

jonaprieto commented on mkdocs_juvix/plugin.py:
```diff
@@ -82,10 +82,26 @@ def on_page_read_source(self, page: Page, config):
+    def on_nav(self, nav: Navigation, config: Config, files: Files):
+
+        for page in nav.pages:
+           if page.file.src_path.endswith("index.juvix.md"):
```
We need to verify that the path is within the docs folder to consider it homepage, as it could be in a different subfolder.

agureev committed:
Add Support for README.juvix.md as Homepage

Allows treating `README.juvix.md` as a homepage in the absence of an
index file.

agureev committed:
Update plugins

Enables index and readme file identification through
equalities. Removes `on_nav` as an unnessary plugin.
